# Masterproject-Petrotoga

In this repository are found the scripts used as part of the comparative genomic analysis of the genus _Petrotoga_.
The analysis was mainly based on preexisting software applications such as [stand-alone BLAST](https://www.ncbi.nlm.nih.gov/books/NBK52640/) ([1](https://github.com/matildecoin/masterproject-Petrotoga/wiki/Bibliography)), [Count](http://www.iro.umontreal.ca/~csuros/gene_content/count.html) ([2](https://github.com/matildecoin/masterproject-Petrotoga/wiki/Bibliography)), [Prokka](https://github.com/tseemann/prokka) ([3](https://github.com/matildecoin/masterproject-Petrotoga/wiki/Bibliography)), and [Roary](https://sanger-pathogens.github.io/Roary/) ([4](https://github.com/matildecoin/masterproject-Petrotoga/wiki/Bibliography)).

The project aims to conduct a comparative genomics analysis of the seven cultivated and sequenced _Petrotoga_ species, along with their closest phylogenetic neighbors, to identify genomic signatures associated with their extreme living environment. 

The analysis is divided in three parts: 

i)  [Specie and Genus boundaries](https://github.com/mcmcoin/masterproject-Petrotoga/wiki/Specie-and-Genus-determination).

ii) [Comparative analysis based on pangenome](https://github.com/mcmcoin/masterproject-Petrotoga/wiki/Pangenome-analysis).
Functional analysis of core and specific genes, identification of marker genes.

iii) [Reconstruction of the evolutionary history](https://github.com/mcmcoin/masterproject-Petrotoga/wiki/Evolutionary-history-reconstruction).
Especially of pathways of interest.

A more detailed description of the analysis can be found in the Wiki.
